### DISCUSSION TITLE
Impossible to clear Event_logs on busy server

### DISCUSSION THREAD

**Discussion by ySp-chld** [dani-garcia/vaultwarden#4802] [Q&A] [answered]
Using Vaultwarden 1.31.0

I have log events for 90 days enable, with the popularity of password manager our instance came to have a constant flow of people using the API and the service in the org.
The constant flow makes the DB 15GB now and basically any attempt to reduce the size of the event table has failed.
And the backup of our poor sqlite is failling because the activity is non stop apparently.

I've tried to reduce the amount of day in the table but it fails, I've try to disable the event logs but it does clear the table.  

I've tried to reduce the number of day gradually but it seems that sometime the job just does nothing. 
I can see the entry :  

`Rocket has launched from http://0.0.0.0:80
Start events cleanup job
` 
But sometime it feels like it's doing something because I can see some `.wal` file in the sqlite but the activity is so intense that it cannot manage to clear anything and just stall .

I'm very sorry if this is not the proper place to report but I think there's several bug in the event org feature : 

 - to my knowledge there's no way to force a job clean up, from API, UI or CLI.
 - there's after looking at the code no message indicating the end of the job clean up.

I know we should moove our setup to anything but sqlite at that scale this is in the pipeline, in the mean time, as a manual clean up  can I safely shutdown event logs and drop the table manually from CLI and then reenable them with a smaller retention ? 
Or is there an other option I don't know about.

PS: Sorry if I report in the wrong place, I feel like it belongs here, feel free to close if I should use forum instead.
Also thank you very much to every one who work on this amazing project. :heart: 

PS2: This is very much a SQLite issue, I need to run the Vacuum. the cleanup actually worked but never showed as the DB is very fragmented now.

**Comment by BlackDex** [comment, marked as answer]
I think you need to read a bit more how SQLite works.
Those records are deleted, but the location is kept there and thus you will not see the filesize to shrink.
SQLite will reuse those record locations to store new data.
This is just how SQLite works.

We are not going to build-in an automatic vacuum option, since that might cause issues if someone wants to do recovery actions on the database.

**Comment by ySp-chld** [comment]
I indeed need to learn more about sqlite, and I think we will moove to a Postgres or mariaDB anyway. 
I'm currently doing a manual vacuum and will reduce the log retention to insure that our backup can run.
Thank you for taking the time to answer :-)

**Reply by BlackDex** [reply]
btw, if you realy want, you could set auto_vacuum on.

1. Shutdown Vaultwarden
2. Open the database with the sqlite3 cli client `sqlite3 db.sqlite3`
3. Run `PRAGMA auto_vacuum = '1'; VACUUM;` and then `.quit`
4. Start Vaultwarden again.

This might help you to cleanup storage, but it will probably fill-up again rather soon :).

**Reply by BlackDex** [reply]
Also, MariaDB and PostgreSQL behave the same way. They will not cleanup those records, and also reuse the previous records space as far as i know.

**Reply by ySp-chld** [reply]
Yes, but the underlying for now is that sqllite .backup never manage to finish of takes an absurd amount of time because there's a constant flow of new logs (that's a supposition)  So If I can reduce the amount of event log to few days I hope it will help us to backup vaultwarden.

**Reply by BlackDex** [reply]
Ah, yea that might help.
Also, not sure if it will help to not use `.backup` but rather use `VACUUM INTO /path/to/backup.sqlite3`.

**Comment by ySp-chld** [comment]
As a side note I still think that the absence of message to notify of the end of the cleanup process is a small problem :-)
